src/engine/GroupBy.cpp

```diff
@@ -795,7 +810,7 @@ bool GroupBy::findAggregatesImpl(
   }
 
   // Unsupported aggregates
-  if (!isSupportedAggregate(expr)) return false;
+  if (isUnsupportedAggregate(expr)) return false;
```

Comment:
If you move this check to the beginning (before checking for AVG), then the coverage gets more happy.